nsds_lab_to_nwb/components/stimulus/stim_value_extractor.py

```diff
@@ -2,67 +2,116 @@
 import os
 import numpy as np
 import csv
-import h5py
 
-# import pkg_resources
+from nsds_lab_to_nwb.common.io import read_mat_file
+from nsds_lab_to_nwb.metadata.stim_name_helper import check_stimulus_name
 
 
 class StimValueExtractor():
-    def __init__(self, stim_values_command, stim_lib_path):
-        self.stim_values_command = stim_values_command
+    def __init__(self, stim_configs, stim_lib_path):
+        self.stim_name, self.stim_info = check_stimulus_name(stim_configs['name'])
+        self.stim_type = stim_configs['type']
+        self.stim_values_command = stim_configs.get('stim_values', None)
         self.stim_lib_path = stim_lib_path
 
     def extract(self):
-        stim_values_command = self.stim_values_command
-        if 'tone_stimulus_values' in stim_values_command:
-            extractor, filename = self.__parse_command(stim_values_command)
-            if extractor != 'tone_stimulus_values':
-                raise ValueError('parsing error')
-            stim_values = eval('{}(\'{}\')'.format(extractor,
-                            os.path.join(self.stim_lib_path, filename)))
-        if 'timit_stimulus_values' in stim_values_command:
-            extractor, filename = self.__parse_command(stim_values_command)
-            if extractor != 'timit_stimulus_values':
-                raise ValueError('parsing error')
-            stim_values = eval('{}(\'{}\')'.format(extractor,
-                            os.path.join(self.stim_lib_path, filename)))
-        if 'gen_tone_stim_vals' in stim_values_command:
-            extractor, _ = self.__parse_command(stim_values_command)
-            if extractor != 'gen_tone_stim_vals':
-                raise ValueError('parsing error')
-            stim_values = eval('{}()'.format(extractor))
-        if 'np.ones' in stim_values_command:
-            stim_values = eval(stim_values_command)
-
-        return stim_values
-
-    def __parse_command(self, command):
-        ''' return (a_a_a, b.b.b) by parsing string 'a_a_a(b.b.b)' '''
-        res = re.match('(\S+)\((\S+)\)', command)
-        return (res.group(1), res.group(2))
+        if self.stim_type == 'continuous':
```

Comment:
It's probably fine for now, but I'm not sure that continuous will always mean no stim labels in the future.

Reply:
Yeah, for now I was simply making use of what's already available in the stimulus metadata. But we can think about a better labeling system